src/Symfony/Component/Validator/ConstraintViolation.php

```diff
@@ -79,17 +79,32 @@ public function __toString()
         }
 
         $propertyPath = (string) $this->propertyPath;
-        $code = $this->code;
 
         if ('' !== $propertyPath && '[' !== $propertyPath[0] && '' !== $class) {
             $class .= '.';
         }
 
-        if (!empty($code)) {
-            $code = ' (code '.$code.')';
+        $context = '';
+        if (!empty($this->code)) {
+            $context .= ' code '.$this->code.', ';
+        }
+        $context .= 'invalid value ';
+        if (\is_object($this->invalidValue)) {
+            // `preg_replace()` is required here to remove "\x00" binary string
+            // from `get_class()` return value when its argument is an anonymous
+            // class ("class@anonymous\x00")
```

Comment:
this case should be handled as in other similar places, eg
```
$class = \get_class($this->invalidValue);
$context .= 'c' === $class[0] && 0 === strpos($class, "class@anonymous\0") ? get_parent_class($class).'@anonymous' : $class;
```

Reply:
Updated in order to use this approach. Thank you!

Reply:
It's fine to use "@anonymous" instead of "class@anonymous" for the dumped value?

Reply:
Sure, that's how we do everywhere else where we need similar friendly display.
BTW $class was fine as a bar name :)
Should at least be $invalidObjectClass (not "Name")